### PR TITLE
Fix misspellings in method names and deprecate old ones

### PR DESCRIPTION
**Pull request by akiojalehto**
### Description
Fixing typos and deprecating misspelled methods

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Eav/Block/Adminhtml/Attribute/Edit/Js.php [hunk 1]
```diff
@@ -36,12 +36,21 @@ public function __construct(
         parent::__construct($context, $data);
     }
 
+    /**
+     * @deprecated Misspelled method
+     * @see getCompatibleInputTypes
```
Comment (orlangur) [inline app/code/Magento/Eav/Block/Adminhtml/Attribute/Edit/Js.php:41]: Same question as https://github.com/magento/magento2/pull/13301/files#r163084396

Comment (akiojalehto) [reply app/code/Magento/Eav/Block/Adminhtml/Attribute/Edit/Js.php:41]: I was pointed out in [previous pr](https://github.com/magento/magento2/pull/13187#discussion_r161676584) to use that approach and it's working great in IDEs, using phpstorm my self.

**Comment by akiojalehto** [comment]
@orlangur Any progress on these, are the changes ok?

**Comment by orlangur** [comment]
@akiojalehto no changes needed, just that I didn't have time to check how it plays with IDE and not always have VPN available. Will finish now.

**Comment by akiojalehto** [comment]
@orlangur Great!

**Review by orlangur — APPROVED** [review]

**Pull request merged**